### DISCUSSION TITLE
Build fails while downloading p2.index from download.eclipse.org

### DISCUSSION THREAD

**Discussion by learnsomethingtoday** [eclipse-tycho/tycho#3298] [Q&A]
I am trying to build **eclipse-mat** from source code using Tycho 4.0.3 and Maven, which fails while getting p2.index from repository, 

Error :
`[INFO] Downloading from p2: https://download.eclipse.org/eclipse/updates/4.30/p2.index`

`[ERROR] Failed to resolve target definition file: eclipse-mat/org.eclipse.mat/org.eclipse.mat.targetdef/mat-2023-12.target: Failed to load p2 metadata repository from location https://download.eclipse.org/eclipse/updates/4.30/: Unable to read repository at https://download.eclipse.org/eclipse/updates/4.30. HTTP connect timed out -> [Help 1]`

I didn't get any working solution for this yet, please provide some help on this.

Thanks in advance!

**Comment by laeubi** [comment]
If you have problems with Eclipse Services you need to report them here:
https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues

beside that make sure you have no special limitations likes proxy or firewalls in place that probably need consideration:
https://maven.apache.org/guides/mini/guide-proxies.html

**Reply by learnsomethingtoday** [reply]
The website is working from my local browser, able to connect to it and view the p2.index file.

But the same thing is not happening on a hosted gitlab runner.. (HTTP connect timed out error)

**Reply by learnsomethingtoday** [reply]
I have set the proxies as per configuration in my settings.xml file too..

**Comment by learnsomethingtoday** [comment]
Found out the issue, was able to pass through the above error.
It was mainly due to a bug in tycho i would say, refer this : https://github.com/eclipse-tycho/tycho/pull/3303

It was failing if I have both of the http and https proxies configured in my file.
As @Bananeweizen suggested to use a single entry of proxy (http) instead of both http and https, it really worked in my case.

Thanks for the help @Bananeweizen.
